File: Programmers/Lv1/Lv1_소수찾기.py
def solution(n):
    answer = 0
    check = [False] * (n+1);
    check[0] = check[1] = True;

    for i in range(2,n+1):
        if check[i] == False:
            for j in range(i*i, n+1, i):
                check[j] = True

    for i in range(2,n+1):
        if check[i] == False:
            answer += 1

    return answer

File: Programmers/Lv1/Lv1_소수찾기.cpp
#include <string>
#include <vector>
using namespace std;

int solution(int n) {
	vector<int> p(1000001);
	vector<bool> check(1000001);
	int pn = 0;
	int answer = 0;
	check[0] = check[1] = true;
	for (int i = 2; i*i <= n; i++) {
		if (check[i] == false) {
			p[pn++] = i;
			for (int j = i + i; j <= n; j += i)
				check[j] = true;
		}
	}

	for (int i = 2; i <= n; i++) {
		if (check[i] == false)
			answer++;
	}

	return answer;	
}
